src/dotnet/commands/dotnet-new/Program.cs

```diff
@@ -19,6 +19,12 @@ private static string GetFileNameFromResourceName(string s)
                 return null;
             }
 
+            // filename.extension.template
+            if (parts.Length > 2 && string.Equals("template", parts[parts.Length - 1], StringComparison.OrdinalIgnoreCase))
```

Comment:
Why not extract this into a method called RemoveTemplateFromEndOfName and remove the comment?